model/riscv_sys_regs.sail

```diff
@@ -180,6 +180,8 @@ function haveUsrMode() -> bool = misa.U() == 0b1
 function haveNExt()    -> bool = misa.N() == 0b1
 /* see below for F and D extension tests */
 
+function haveZmmul()    -> bool = true
```

Comment:
As with the bitmanip PR, keep the lists of dummy Z* extension checks sorted, i.e put Zm* below Zk*

Reply:
> Does M not imply Zmmul, and thus only the latter needs checking?

Yes, M does not imply Zmmul. You can read this discussion on this [PR](https://github.com/riscv-software-src/riscv-isa-sim/pull/830)

Reply:
> As with the bitmanip PR, keep the lists of dummy Z* extension checks sorted, i.e put Zm* below Zk*

Sure :)

Reply:
> > Does M not imply Zmmul, and thus only the latter needs checking?
> 
> Yes, M does not imply Zmmul. You can read the discussion on this [PR](https://github.com/riscv-software-src/riscv-isa-sim/pull/830)

This contradicts https://github.com/riscv/riscv-isa-manual/pull/648#issuecomment-842461775, which is as close to an authoritative source as I can find. Pointing at your own comments on your own Spike pull request is hardly authoritative.

Reply:
Let's put it in this way. An implementation can have both extensions as a time. When I said does not imply I meant to say If misa.M() ==1 it does not mean that Zmmul extension is there as well. if Either or both of them are in implementation the Mul instructions will work. That's why there is a `or` between haveMulDiv and haveZmmul.

Reply:
That doesn't make much sense when Zmmul is defined as a subset of M...

Reply:
As far as I understand. The benefit of Zmmul is, as it supports only MUL instructions (Not DIV/REM), it will save hardware in those implementations where DIV/REM instructions are not needed.

Reply:
One difference between M and Zmmul is that the latter is not controlled by misa.M. If you have Zmmul then the multiply instructions are always enabled. If you have M then the multiply instructions are enabled iff misa.M==1. That's why saying "M implies Zmmul" is inaccurate -- because that would mean that misa.M only controls divide and not multiply.

This Sail model doesn't support modifying misa.M so this is moot here -- but that might change in the future so I think it's better to keep them separate here.

Reply:
Ah indeed that is an important but subtle difference. I believe modifying misa.M should work if sys_enable_writable_misa is true.

Reply:
It looks like [this code](https://github.com/riscv/sail-riscv/blob/4508059032ecfdb986a5673108b02900d2c4ccfd/model/riscv_sys_regs.sail#L157-L168) supports modifying bits C/D/F only. And reading it, I see a bug, which I'll file separately.

Reply:
Oh, right, I skimmed it and thought it was mostly fixing up v, but it's just using v to fix up the existing m.